### PR TITLE
Remove pubsub and sockets

### PR DESCRIPTION
**Pull request by kevinburkeshyp**
They're unused and we don't want them.

### CODE REVIEW

**Comment by benbuckman** [comment]
Wow, Sails is a beast.

**Comment by caseycrites** [comment]
Why not just remove the `lib/hooks/pubsub` directory?

**Comment by caseycrites** [comment]
Also, looks like there's a couple lingering references:

```
marmaduke ~/dev/sails[no-sockets=]% grep -r 'pubsub' lib
lib/hooks/blueprints/actions/add.js:      if (!isDuplicateInsertError && req._sails.hooks.pubsub) {
lib/hooks/orm/index.js:            // in case something needs to respond to the ORM reload (e.g. pubsub hook)
lib/hooks/pubsub/README.md:// Any of the pubsub methods..
```

**Comment by caseycrites** [comment]
and a few to `isSocket`:

```
marmaduke ~/dev/sails[no-sockets=]% grep -r 'isSocket' lib
lib/hooks/blueprints/actionUtil.js:      var jsonpOpts = req.options.jsonp && !req.isSocket;
lib/hooks/blueprints/actionUtil.js:    var jsonpOpts = req.options.jsonp && !req.isSocket;
```

**Comment by caseycrites** [comment]
and just some references to `sockets` in general, mostly (only?) documentation:

```
marmaduke ~/dev/sails[no-sockets=]% grep -ir 'sockets' lib
lib/app/request.js:  // (i.e. merge w/ lib/router/req.js and lib/hooks/sockets/lib/interpreter/*.js)
lib/EVENTS.md:> from the built-in router and listen for the routes itself using `app.use`.  On the other hand, in the `sockets` hook,
lib/hooks/http/initialize.js:        // TODO: investigate sharing http middleware with sockets hook..?
lib/hooks/pubsub/README.md:sails.sockets.emit()
lib/hooks/request/metadata.js: * (fails silently if http or sockets is not enabled)
lib/hooks/request/metadata.js:  // TODO: try to use `sockets` if `http` is not available
lib/router/bindDefaultHandlers.js:      // it should monitor for `abort` events and manage its private resources (e.g. TCP sockets) accordingly.
lib/router/bindDefaultHandlers.js:      // it should monitor for `abort` events and manage its private resources (e.g. TCP sockets) accordingly.
lib/router/index.js: * `sockets` hooks for examples of attaching a server to
lib/router/README.md:It is not involved with HTTP, WebSockets, or other internet protocols directly-- instead, it emits
lib/router/README.md:On the other hand, the `sockets` hook defers to the core router, emitting a `request` event whenever
lib/router/req.js: * middleware.  Used by hooks (i.e. sockets) but also for
lib/router/req.js:    // TODO: mimic the session store impl in sockets hook
```

File: npm-shrinkwrap.json [hunk 1]
```diff
@@ -889,53 +889,6 @@
         }
       }
     },
-    "socket.io": {
```
Comment (caseycrites) [inline npm-shrinkwrap.json:892]: :)

**Comment by caseycrites** [comment]
And some leftover config in the test directory:

```
test/integration/fixtures/sampleapp/config/local.js:    sockets: {
```

**Comment by caseycrites** [comment]
There's a couple more small things, again mostly documentation, if you just grep for `socket` instead of the plural. I trust you can turn those up without me pasting them here.

This is awesome, minus 8000 lines and 2 hooks!

**Comment by kevinburkeshyp** [comment]
Alright, the latest 4 commits remove a lot more pubsub/sockets related code

**Pull request merged**